monix-tail/shared/src/test/scala/monix/tail/IterantTakeEveryNthSuite.scala

```diff
@@ -0,0 +1,121 @@
+    assertEquals(naiveImp(iter, 3).toListL.value, List(3, 6))
+    assertEquals(naiveImp(iter, input.length + 1).toListL.value, List.empty[Int])
+  }
+
+  test("Iterant[Task].takeEveryNth equivalence with naiveImp") { implicit s =>
+    check3 { (list: List[Int], idx: Int, nr: Int) =>
+      val stream = arbitraryListToIterant[Task, Int](list, math.abs(idx) + 1, allowErrors = false)
+      val length = list.length
+      // scale down from (Int.MinValue to Int.MaxValue) to (1 to (length + 1)) range
+      val n = math.round(
+        (length * (nr.toDouble - Int.MinValue.toDouble)) / (Int.MaxValue.toDouble - Int.MinValue.toDouble) + 1
+      ).toInt
+      val actual = stream.takeEveryNth(n).toListL.runAsync
+      val expected = naiveImp(stream, n).toListL.runAsync
+      s.tick()
+      actual.value <-> expected.value
+    }
+  }
+
+  test("Iterant.takeEveryNth protects against broken batches") { implicit s =>
+    check1 { (iter: Iterant[Task, Int]) =>
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Task].nextBatchS[Int](new ThrowExceptionBatch(dummy), Task.now(Iterant[Task].empty), Task.unit)
+      val stream = iter.onErrorIgnore ++ suffix
+      val received = stream.takeEveryNth(1)
+      received <-> Iterant[Task].haltS[Int](Some(dummy))
+    }
+  }
+
+  test("Iterant.takeEveryNth protects against broken cursors") { implicit s =>
+    check1 { (iter: Iterant[Task, Int]) =>
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Task].nextCursorS[Int](new ThrowExceptionCursor(dummy), Task.now(Iterant[Task].empty), Task.unit)
+      val stream = iter.onErrorIgnore ++ suffix
+      val received = stream.takeEveryNth(1)
+      received <-> Iterant[Task].haltS[Int](Some(dummy))
+    }
+  }
+
+  test("Iterant.takeEveryNth triggers early stop on exception") { _ =>
+    check1 { (iter: Iterant[Coeval, Int]) =>
+      val cancelable = BooleanCancelable()
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Coeval].nextCursorS[Int](new ThrowExceptionCursor(dummy), Coeval.now(Iterant[Coeval].empty), Coeval.unit)
+      val stream = (iter.onErrorIgnore ++ suffix).doOnEarlyStop(Coeval.eval(cancelable.cancel()))
+      intercept[DummyException] { stream.takeEveryNth(1).toListL.value }
+      cancelable.isCanceled
+    }
+  }
+
+  test("Iterant.takeEveryNth preserves the source earlyStop") { implicit s =>
+    var effect = 0
+    val stop = Coeval.eval(effect += 1)
+    val source = Iterant[Coeval].nextCursorS(BatchCursor(1,2,3), Coeval.now(Iterant[Coeval].empty[Int]), stop)
+    val stream = source.takeEveryNth(1)
+    stream.earlyStop.value
+    assertEquals(effect, 1)
+  }
+
+  test("Iterant.takeEveryNth throws on invalid n") { implicit s =>
+    val source = Iterant[Coeval].nextCursorS(BatchCursor(1,2,3), Coeval.now(Iterant[Coeval].empty[Int]), Coeval.unit)
+    intercept[IllegalArgumentException] { source.takeEveryNth(0).toListL.value }
```

Comment:
Just a note — we need to be precise about what throws the error. Given you have a `require` in `takeEveryNth`, then you should delete `.toListL.value`.